generator/lib/code_builders/assignment_builder.dart

```diff
@@ -50,16 +51,31 @@ Expression generateSourceFieldAssignment(SourceAssignment sourceAssignment,
         expr = refer(matchingMappingListMethods.first.name);
       }
 
-      sourceFieldAssignment =
-          // source.{field}.map
-          sourceReference
-              .property(sourceField.name)
-              .property('map')
-              // (expr)
-              .call([expr])
-              //.toList()
-              .property('toList')
-              .call([]);
+      if (sourceField.type.isDartCoreList &&
+          sourceField.type.nullabilitySuffix == NullabilitySuffix.question) {
+        sourceFieldAssignment =
+            // source.{field}.map
+            sourceReference
+                .property(sourceField.name)
+                .nullChecked
```

Comment:
This has been added to add `!`. Due to this, I had to pub upgrade the dependencies.